### PR TITLE
S3bucket

### PR DESCRIPTION
**Pull request by aibunny**
# Description
I added the s3 bucket for persistent images but i left out the access and secret keys part keys part for later when we have an amazon account for the project.

### CODE REVIEW

**Review by JimmyTron — COMMENTED** [review]
**Expected Changes**

1. Requirments:  `Install boto3 and django-storages packages using pip.`
2. configuration to your project's urls.py file to serve static and media files from S3: 
Screenshot from 2023-03-10 04-02-19

## The changes are spanning beyond the scope of the pull request which is not a good thing. Try to only implement a single feature per pull request. It make reviews easier and reduces chances of introducing bugs.

> For instance on this we have tree unrelated fixes:

1. `Reactions`
2. `S3 storage`
3. `Readme Update`

File: CMS/settings.py [hunk 1]
```diff
@@ -224,3 +226,12 @@
...
 USER_PASSWORD_LENGTH = 8
+
+# boto3 to connect to S3 bucket
+DEFAULT_FILE_STORAGE = 'storages.backends.s3boto3.S3Boto3Storage'
+
+AWS_QUERY_STRING = False
+AWS_ACCESS_KEY_ID =os.getenv('AWS_ACCESS_KEY_ID')
+AWS_SECRET_ACCESS_KEY =os.getenv('AWS_SECRET_ACCESS_KEY')
+AWS_STORAGE_BUCKET_NAME =os.getenv('AWS_STORAGE_BUCKET_NAME')
+AWS_QUERYSTRING_AUTH = False
```
[the remark this answers is not in the document]
Comment (aibunny) [reply CMS/settings.py:237]: the settings in my pr are for a public bucket the other settings are made during the creating of the bucket in the s3 console as part of the bucket policy. Remember we're using this bucket for images which are supposed to be publicly uploaded and read by anyone using the cms, so we don't actually need ``` AWS_DEFAULT_ACL = None```. But if we want to upload our static files to s3 bucket the we can make the bucket private or the particular folder for images and videos  can be set to public read

File: blog/models.py [hunk 2]
```diff
@@ -56,23 +64,25 @@ class Comment(models.Model):
...
+        return 'Comment {} by {}'.format(self.body,self.user.account_name) 
+    
+    def child_comment(self):#get all child comments
+        return Comment.objects.filter(parent_comment=self)
+    
+    @property #make comment a parent if it gets a reply 
+    def is_parent(self):
+        if self.parent_comment is not None:
+            return False
+        return True
```
[the remark this answers is not in the document]
Comment (aibunny) [reply blog/models.py:83]: It doesn't appear as changed its part of the comments

File: blog/models.py [hunk 3]
```diff
@@ -56,23 +64,25 @@ class Comment(models.Model):
     Post_id = models.ForeignKey(Stori,on_delete= models.CASCADE)
     user = models.ForeignKey(Account, on_delete= models.CASCADE)
     date_created = models.DateTimeField(auto_now_add=True)
-    reactions = models.PositiveIntegerField(default=0)
+    reactions = GenericRelation(Reaction)
+    #make the initial comment parent
+    parent_comment = models.ForeignKey('self',on_delete=models.CASCADE,blank=True,null=True,related_name='replies')
```
[the remark this answers is not in the document]
Comment (JimmyTron) [reply blog/models.py:69]: Im yet to confirm this though

**Review by JimmyTron — APPROVED** [review]

**Review by JimmyTron — CHANGES_REQUESTED** [review]
Requirements missing 
> check also pull request #36

**Review by JimmyTron — DISMISSED** [review]

**Review by Rioba-Ian — APPROVED** [review]

**Pull request closed without merging**